### PR TITLE
feat: add support for Hyperaccel LPU devices

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR updates webui to support querying, altering and attaching Hyperaccel LPU devices.

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

File: resources/device_metadata.json [hunk 1]
```diff
@@ -98,6 +98,17 @@
...
+    "lpu": {
+      "slot_name": "lpu.device",
+      "description": "Hyperaccel LPU",
+      "human_readable_name": "LPU Device",
+      "display_unit": "LPU",
+      "number_format": {
+        "binary": false,
+        "round_length": 0
+      },
+      "display_icon": "furiosa"
```
Comment (inureyes) [inline resources/device_metadata.json]: Need to change icon?

**Review by inureyes — CHANGES_REQUESTED** [review]
Device icon needs to change from furious to hyperaccel icon.

File: react/src/components/ResourceNumber.tsx [hunk 2]
```diff
@@ -140,6 +142,7 @@ export const ResourceTypeIcon: React.FC<AccTypeIconProps> = ({
     'ipu.device': [<MWCIconWrap size={size}>view_module</MWCIconWrap>, 'IPU'],
     'atom.device': ['/resources/icons/rebel.svg', 'ATOM'],
     'warboy.device': ['/resources/icons/furiosa.svg', 'Warboy'],
+    'lpu.device': ['/resources/icons/furiosa.svg', 'LPU'],
```
Comment (rapsealk) [inline react/src/components/ResourceNumber.tsx]:
```suggestion
    'lpu.device': ['/resources/icons/hyperaccel.svg', 'LPU'],
```

File: react/src/hooks/useResourceLimitAndRemaining.tsx [hunk 3]
```diff
@@ -267,6 +267,7 @@ export const useResourceLimitAndRemaining = ({
             'ipu.device': 'maxIPUDevicesPerContainer',
             'atom.device': 'maxATOMDevicesPerContainer',
             'warboy.device': 'maxWarboyDevicesPerContainer',
+            'lpu.device': 'maxWarboyDevicesPerContainer', // FIXME: add maxWarboyDevicesPerContainer to config
```
Comment (rapsealk) [inline react/src/hooks/useResourceLimitAndRemaining.tsx]:
```suggestion
            'lpu.device': 'maxLPUDevicesPerContainer', // FIXME: add maxLPUDevicesPerContainer to config
```

File: src/components/backend-ai-agent-list.ts [hunk 4]
```diff
@@ -1188,6 +1206,30 @@ export default class BackendAIAgentList extends BackendAIPage {
...
             : html``}
+          ${rowData.item.lpu_slots
+            ? html`
+                <div
+                  class="layout horizontal center-justified flex progress-bar-section"
+                >
+                  <div class="layout horizontal start resource-indicator">
+                    <img
+                      class="indicator-icon fg green"
+                      src="/resources/icons/furiosa.svg"
```
Comment (rapsealk) [inline src/components/backend-ai-agent-list.ts]:
```suggestion
                      src="/resources/icons/hyperaccel.svg"
```

File: src/components/backend-ai-agent-summary-list.ts [hunk 5]
```diff
@@ -718,6 +736,30 @@ export default class BackendAIAgentSummaryList extends BackendAIPage {
...
             : html``}
+          ${rowData.item.lpu_slots
+            ? html`
+                <div
+                  class="layout horizontal center-justified flex progress-bar-section"
+                >
+                  <div class="layout horizontal start resource-indicator">
+                    <img
+                      class="indicator-icon fg green"
+                      src="/resources/icons/tpu.svg"
```
Comment (rapsealk) [inline src/components/backend-ai-agent-summary-list.ts]:
```suggestion
                      src="/resources/icons/hyperaccel.svg"
```

File: src/components/backend-ai-session-launcher-neo.ts [hunk 6]
```diff
@@ -2685,6 +2698,59 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+            if (
+              parseInt(lpu_device_metric.max) !== 0 &&
+              lpu_device_metric.max !== 'Infinity' &&
+              !isNaN(lpu_device_metric.max) &&
+              lpu_device_metric.max != null
+            ) {
+              lpu_device_metric.max = Math.min(
+                parseInt(lpu_device_metric.max),
+                parseInt(this.userResourceLimit['lpu.device']),
+                available_slot['cuda_device'],
```
Comment (rapsealk) [inline src/components/backend-ai-session-launcher-neo.ts]:
```suggestion
                available_slot['lpu_device'],
```

File: src/components/backend-ai-session-launcher-neo.ts [hunk 7]
```diff
@@ -2685,6 +2698,59 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+                this.max_cuda_device_per_container,
+              );
+            } else {
+              lpu_device_metric.max = Math.min(
+                parseInt(this.userResourceLimit['lpu.device']),
+                parseInt(available_slot['cuda_device']),
```
Comment (rapsealk) [inline src/components/backend-ai-session-launcher-neo.ts]:
```suggestion
                parseInt(available_slot['lpu_device']),
```

File: src/components/backend-ai-session-launcher.ts [hunk 8]
```diff
@@ -2688,6 +2701,59 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+              lpu_device_metric.max = Math.min(
+                parseInt(lpu_device_metric.max),
+                parseInt(this.userResourceLimit['lpu.device']),
+                available_slot['cuda_device'],
+                this.max_cuda_device_per_container,
+              );
+            } else {
+              lpu_device_metric.max = Math.min(
+                parseInt(this.userResourceLimit['lpu.device']),
+                parseInt(available_slot['cuda_device']),
```
Comment (rapsealk) [inline src/components/backend-ai-session-launcher.ts]:
```suggestion
                parseInt(available_slot['lpu_device']),
```

File: src/components/backend-ai-session-launcher.ts [hunk 9]
```diff
@@ -2688,6 +2701,59 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+            if (
+              parseInt(lpu_device_metric.max) !== 0 &&
+              lpu_device_metric.max !== 'Infinity' &&
+              !isNaN(lpu_device_metric.max) &&
+              lpu_device_metric.max != null
+            ) {
+              lpu_device_metric.max = Math.min(
+                parseInt(lpu_device_metric.max),
+                parseInt(this.userResourceLimit['lpu.device']),
+                available_slot['cuda_device'],
```
Comment (rapsealk) [inline src/components/backend-ai-session-launcher.ts]:
```suggestion
                available_slot['lpu_device'],
```

File: src/components/backend-ai-session-list.ts [hunk 10]
```diff
@@ -3187,13 +3192,24 @@ ${rowData.item[this.sessionNameField]}</pre
                     <span class="indicator">Warboy</span>
                   `
                 : html``}
+              ${rowData.item.lpu_slot
+                ? html`
+                    <img
+                      class="indicator-icon fg green"
+                      src="/resources/icons/furiosa.svg"
```
Comment (rapsealk) [inline src/components/backend-ai-session-list.ts]:
```suggestion
                      src="/resources/icons/hyperaccel.svg"
```

File: src/components/backend-ai-settings-view.ts [hunk 11]
```diff
@@ -589,6 +591,27 @@ export default class BackendAiSettingsView extends BackendAIPage {
                       ></mwc-switch>
                     </div>
                   </div>
+                  <div class="horizontal layout setting-item">
+                    <div
+                      class="vertical center-justified layout setting-desc-shrink"
+                    >
```
Comment (lizable) [inline src/components/backend-ai-settings-view.ts:597]: We need translations for `LPUsupport`, `DescLPUsupport`, `RequireLPUPlugin`.

File: src/components/backend-ai-resource-panel.ts [hunk 12]
```diff
@@ -900,6 +912,41 @@ export default class BackendAIResourcePanel extends BackendAIPage {
...
+                                          .lpu_total} LPUs ${_t(
+                                          'summary.reserved',
+                                        )}."
+                                      ></lablup-progress-bar>
+                                      <lablup-progress-bar
+                                        id="lpu-usage-bar-2"
+                                        class="end"
+                                        progress="0"
+                                        description="${_t(
+                                          'summary.LPUEnabled',
```
Comment (lizable) [inline src/components/backend-ai-resource-panel.ts]: Here as well.

File: src/components/backend-ai-settings-view.ts (hunk 11 above)
Comment (yomybaby) [reply src/components/backend-ai-settings-view.ts:597]: image

**Review by lizable — COMMENTED** [review]
Found some missing translations, and minor icon matching. Except that, it's okay to merge.

**Review by lizable — APPROVED** [review]
LGTM.

Btw we'd better add those AI accelerators(including Hyperaccel LPU and others) into resource policy as well.

**Review by lizable — APPROVED** [review]
LGTM.

**Comment by inureyes** [comment]
LGTM. 

We need to continue https://github.com/lablup/backend.ai-webui/pull/1937 ( ref. @lizable @kyujin-cho @agatha197 )

**Review by inureyes — APPROVED** [review]

**Pull request merged**